src/model/code-visualizer.ts

```diff
@@ -0,0 +1,86 @@
+import { WordType, ICode, IWord } from './models';
+
+export class CodeVisualizer {
+  private code: ICode;
+
+  /**
+   * Helper class to help to render the code snippet in vue
+   *
+   * @param code
+   */
+  public constructor(code: ICode) {
+    this.code = code;
+  }
+
+  /**
+   * Useful to enumerate in vue components
+   *
+   * @returns list of lines with list of words
+   */
+  public getCodeLines(): Array<Array<IWord>> {
+    const lines: Array<Array<IWord>> = [];
+    let currentLineWords: Array<IWord> = [];
+    for (let i = 0; i < this.code.words.length; i++) {
+      const wordObj: IWord = this.code.words[i];
+      const word = wordObj.word;
+      if (word == WordType.NEWLINE) {
+        lines.push(currentLineWords);
+        currentLineWords = [];
+      } else if (word == WordType.TAB) {
+        currentLineWords.push(wordObj);
+      } else {
+        currentLineWords.push(wordObj);
+      }
+    }
+    if (currentLineWords.length > 0) {
+      lines.push(currentLineWords);
```

Comment:
Alternatively, also push the empty lines.
Will probably make things easier for a little bit more memory.